JSON parsing error when running server in environment with 2 CPU cores

### Observed behavior

When setting up a new dev environment on a VM machine with just 2 CPU cores, run into JSON parsing error. @rtibbles pointed out it was happening here:

```
const data = JSON.parse(process.env.data);
const index = JSON.parse(process.env.index);
const options = JSON.parse(process.env.options);
const start = JSON.parse(process.env.start);
```
> It's because the JSON parsing is in the outer most scope of the module
https://github.com/learningequality/kolibri/blob/develop/packages/kolibri-tools/lib/webpackdevserver.js#L114
Which means it gets done on module import, not just when the module is run as main (for the non-multi-core run, the module gets imported, but not directly executed, unlike in the multi-core scenario)


### Expected behavior
Functional dev environment even on 2 core devices.

### User-facing consequences
Devs with machines with lower specs unable to run the server.

### Errors and logs

```
(kolibri) didivm@didivm-VirtualBox:~/LE/kolibri$ yarn run devserver
yarn run v1.15.2
$ npm-run-all --parallel django-devserver lint-frontend:watch:format hashi-dev watch
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1556896835008-0.6888841131762105/node but npm is using /home/didivm/.venvs/kolibri/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 watch /home/didivm/LE/kolibri
> kolibri-tools build dev --file ./build_tools/build_plugins.txt

npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1556896835008-0.6888841131762105/node but npm is using /home/didivm/.venvs/kolibri/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 lint-frontend:watch:format /home/didivm/LE/kolibri
> yarn run lint-frontend --monitor --write

npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1556896835008-0.6888841131762105/node but npm is using /home/didivm/.venvs/kolibri/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 django-devserver /home/didivm/LE/kolibri
> kolibri --debug manage runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" 

npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1556896835008-0.6888841131762105/node but npm is using /home/didivm/.venvs/kolibri/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 hashi-dev /home/didivm/LE/kolibri
> yarn workspace hashi run dev

WARNING:root:No C Extensions available for this platform.

$ kolibri-tools lint '{kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}' --ignore '**/node_modules/**','**/static/**','**/packages/kolibri-core-for-export/**' --monitor --write
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
$ yarn run build-base --mode=development --watch
WARNING:root:No C Extensions available for this platform.

INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Version was 0.12.3a0.dev0+git.386.g166d726a, new version: 0.12.3a0.dev0+git.500.gd18cb402
INFO     Running update routines for new version...
$ webpack --config ./webpack.config.js --mode=development --watch
Kolibri Linter: Initializing watcher for the following patterns: {kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}
Operations to perform:
  Apply all migrations: admin, analytics, auth, content, contenttypes, device, discovery, exams, kolibriauth, lessons, logger, morango, notifications, sessions
Running migrations:
  No migrations to apply.

webpack is watching the files…

undefined:1
undefined
^

SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/home/didivm/LE/kolibri/packages/kolibri-tools/lib/webpackdevserver.js:114:19)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at spawnWebpackProcesses (/home/didivm/LE/kolibri/packages/kolibri-tools/lib/cli.js:215:33)
    at Command.<anonymous> (/home/didivm/LE/kolibri/packages/kolibri-tools/lib/cli.js:229:7)
    at Command.listener (/home/didivm/LE/kolibri/node_modules/commander/index.js:315:8)
    at Command.emit (events.js:189:13)
    at Command.parseArgs (/home/didivm/LE/kolibri/node_modules/commander/index.js:654:12)
    at Command.parse (/home/didivm/LE/kolibri/node_modules/commander/index.js:474:21)
    at checkVersion (/home/didivm/LE/kolibri/packages/kolibri-tools/lib/cli.js:342:13)
    at /home/didivm/LE/kolibri/node_modules/check-node-version/index.js:133:7
    at end (/home/didivm/LE/kolibri/node_modules/run-parallel/index.js:18:15)
    at done (/home/didivm/LE/kolibri/node_modules/run-parallel/index.js:22:10)
    at each (/home/didivm/LE/kolibri/node_modules/run-parallel/index.js:28:7)
    at /home/didivm/LE/kolibri/node_modules/run-parallel/index.js:38:43
    at /home/didivm/LE/kolibri/node_modules/check-node-version/index.js:59:14
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kolibri-root@0.12.0-dev.1 watch: `kolibri-tools build dev --file ./build_tools/build_plugins.txt`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/didivm/.npm/_logs/2019-05-03T15_20_40_364Z-debug.log
ERROR: "watch" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
(kolibri) didivm@didivm-VirtualBox:~/LE/kolibri$ 

```

[2019-05-03T15_20_40_364Z-debug.log](https://github.com/learningequality/kolibri/files/3142721/2019-05-03T15_20_40_364Z-debug.log)


### Steps to reproduce

Run the current commands to start the server on a device with 2 CPU cores.

### Context
 * Kolibri version: current `develop` branch on a fresh clone, fresh venv
 * Operating system: Ubuntu Bionic

--- a/packages/kolibri-tools/lib/build.js
+++ b/packages/kolibri-tools/lib/build.js
@@ -1,0 +1,20 @@
+function getEnvVar(variable) {
+  try {
+    return JSON.parse(process.env[variable]);
+  } catch (e) {
+    return process.env[variable];
+  }
+}
+
+function getEnvVars() {
+  return {
+    data: getEnvVar('data'),
+    index: getEnvVar('index'),
+    options: getEnvVar('options'),
+    start: getEnvVar('start'),
+  };
+}
+
+module.exports = {
+  getEnvVars,
+};

--- a/packages/kolibri-tools/lib/bundleStats.js
+++ b/packages/kolibri-tools/lib/bundleStats.js
@@ -3,6 +3,7 @@ const webpack = require('webpack');
 // Import the production configuration so that we remain consistent
 const webpackConfig = require('./production').webpackConfig;
 const logger = require('./logging');
+const { getEnvVars } = require('./build');
 
 const buildLogging = logger.getLogger('Kolibri Build Stats');
 
@@ -25,9 +26,8 @@ function buildWebpack(data, index, startCallback, doneCallback, options) {
   return compiler;
 }
 
-const data = JSON.parse(process.env.data);
-const index = JSON.parse(process.env.index);
-const start = JSON.parse(process.env.start);
+const { data, index, start } = getEnvVars();
+
 function build() {
   buildWebpack(
     data,

--- a/packages/kolibri-tools/lib/i18n.js
+++ b/packages/kolibri-tools/lib/i18n.js
@@ -6,6 +6,7 @@ const os = require('os');
 const webpack = require('webpack');
 const logger = require('./logging');
 const webpackBaseConfig = require('./webpack.config.base');
+const { getEnvVars } = require('./build');
 
 function webpackConfig(pluginData) {
   const pluginBundle = webpackBaseConfig(pluginData);
@@ -30,9 +31,8 @@ function buildWebpack(data, index, startCallback, doneCallback) {
   return compiler;
 }
 
-const data = JSON.parse(process.env.data);
-const index = JSON.parse(process.env.index);
-const start = JSON.parse(process.env.start);
+const { data, index, start } = getEnvVars();
+
 function build() {
   buildWebpack(
     data,

--- a/packages/kolibri-tools/lib/production.js
+++ b/packages/kolibri-tools/lib/production.js
@@ -5,6 +5,7 @@
 const webpack = require('webpack');
 const webpackBaseConfig = require('./webpack.config.base');
 const logger = require('./logging');
+const { getEnvVars } = require('./build');
 
 function webpackConfig(pluginData) {
   const pluginBundle = webpackBaseConfig(pluginData, {
@@ -40,9 +41,8 @@ function buildWebpack(data, index, startCallback, doneCallback) {
   return compiler;
 }
 
-const data = JSON.parse(process.env.data);
-const index = JSON.parse(process.env.index);
-const start = JSON.parse(process.env.start);
+const { data, index, start } = getEnvVars();
+
 function build() {
   buildWebpack(
     data,

--- a/packages/kolibri-tools/lib/webpackdevserver.js
+++ b/packages/kolibri-tools/lib/webpackdevserver.js
@@ -14,6 +14,7 @@ const webpack = require('webpack');
 const openInEditor = require('launch-editor-middleware');
 const webpackBaseConfig = require('./webpack.config.base');
 const logger = require('./logging');
+const { getEnvVars } = require('./build');
 
 const buildLogging = logger.getLogger('Kolibri Webpack Dev Server');
 
@@ -111,10 +112,8 @@ function buildWebpack(data, index, startCallback, doneCallback, options) {
   return compiler;
 }
 
-const data = JSON.parse(process.env.data);
-const index = JSON.parse(process.env.index);
-const options = JSON.parse(process.env.options);
-const start = JSON.parse(process.env.start);
+const { data, index, options, start } = getEnvVars();
+
 function build() {
   buildWebpack(
     data,

